zerver/actions/alert_words.py

```diff
@@ -1,16 +1,16 @@
-from typing import Iterable, Sequence
+from typing import Dict, Iterable, Sequence
 
-from zerver.lib.alert_words import add_user_alert_words, remove_user_alert_words
+from zerver.lib.alert_words import AlertWordData, add_user_alert_words, remove_user_alert_words
 from zerver.models import UserProfile
 from zerver.tornado.django_api import send_event
 
 
-def notify_alert_words(user_profile: UserProfile, words: Sequence[str]) -> None:
+def notify_alert_words(user_profile: UserProfile, words: Sequence[Dict[str, str]]) -> None:
     event = dict(type="alert_words", alert_words=words)
```

Comment:
```suggestion
def notify_alert_words(user_profile: UserProfile, words: Sequence[AlertWordData]) -> None:
    event = dict(
        type="alert_words",
        alert_words=TypeAdapter(Sequence[AlertWordData]).dump_python(words),
    )
```

(Eventually as we migrate the event dicts themselves to Pydantic models, this `TypeAdapter` will become unnecessary here.)